### DISCUSSION TITLE
Self hosted Vector container fails to start

### DISCUSSION THREAD

**Discussion by riccardolardi** [supabase/supabase#26362] [Questions]
I just cloned a fresh copy of supabase, pulled the images and ran `docker compose up` but the Vector container fails to start due to `ERROR vector::config::loading: Error opening config file. error=Permission denied (os error 13) path="etc/vector/vector.yml"`

Running on a VPS `Debian 5.10.216-1 (2024-05-03) x86_64 GNU/Linux` using `Docker v26.1.2, build 211e74b`

Any help is greatly appreciated!

<img width="1443" alt="Screenshot 2024-05-15 at 23 21 34" src="https://github.com/supabase/supabase/assets/422124/e8fdc7ee-feca-4778-88d9-44041ef65586">

**Comment by mpross512** [comment]
For some reason, vector.yml gets created as a directory. You will need to remove vector.yml, and recreate it as a regular file. Then you will need to paste in the contents from [this file](https://github.com/supabase/supabase/blob/master/docker/volumes/logs/vector.yml) in the Supabase GitHub. Lastly, you will need to remove the old supabase-vector docker volume, so docker doesn't complain that you're trying to mount a file to a directory. This can be done by running `docker rm -v supabase-vector`

**Reply by shinnenkara** [reply]
Hi @mpross512! How exactly can it be done? Do I need to somehow change the image that container trying to run?
I am new into docker, so please excuse me if it is the obvious question. But I would appreciate the directions I need to look into.

**Reply by shinnenkara** [reply]
I got this when trying to run the [timberio/vector:0.28.1-alpine](app://dd/dashboard/images/local/details?image=timberio%2Fvector%3A0.28.1-alpine&imageId=sha256%3Ae1dceaa760d264d5d1325f8852aa2e49e536ced4be35c54d02758d0dd7f29f45) image
```
2024-08-26 16:15:57 2024-08-26T23:15:57.697925Z  INFO vector::app: Internal log rate limit configured. internal_log_rate_secs=10
2024-08-26 16:15:57 2024-08-26T23:15:57.697998Z  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info,buffers=info,lapin=info,kube=info"
2024-08-26 16:15:57 2024-08-26T23:15:57.698043Z  INFO vector::app: Loading configs. paths=["etc/vector/vector.yml"]
2024-08-26 16:15:57 2024-08-26T23:15:57.698625Z ERROR vector::cli: Configuration error. error=Is a directory (os error 21)
```

**Reply by Erudition** [reply]
This was it for me.  In fact every time an individual file was bound it gets created as a directory. Thank you. I was forced to git clone the supabase repo and use its `docker/` folder as the base for my `volumes/`.

**Comment by riccardolardi** [comment]
I forgot to follow up - my issue was that the hosting machine had globally enabled usernamespace remapping for all containers https://docs.docker.com/engine/security/userns-remap which would not allow for bind mounts. Would be interesting to know if there was a best practice way to deploy supabase on such environments

**Comment by HTA86** [comment]
I encountered a similar issue and resolved it by changing the Docker Compose configuration. Instead of mounting the individual vector.yml file, I mounted the entire directory:

```yml
- ./volumes/logs/:/etc/vector/:ro
```
This approach avoids issues with file permissions and directory mismatches.

Hope this helps!

**Comment by grigio** [comment]
Thanks, I'm still having a similar error with vector.yml

**Comment by SiJBC** [comment]
I removed the vector database - guess the vector db is part of supabase secret sauce

**Reply by leontepe** [reply]
don't think so, this is probably just a bug

**Comment by Jens-EstateMate** [comment]
Using the CLI to start supabase I have excluded vector from the startup (until such time that they get around to fix this).

`npx supabase start --exclude vector`

Hope this helps someone

**Reply by vicXvac** [reply]
thanks a lot, this helped

**Comment by plastic-shurikens** [comment]
In my case `volumes/logs/vector.yml` file was a directory, uninstall/install supabase fixed it for me 

Compose down the docker.yml 
`docker compose down`

Remove any related volumes
`rm -rf volumes/db/data/`
`docker volume ls`
`docker volume rm <supabase_volume_name>`

Clear unused docker resources
`docker system prune -a`

Then docker compose up again without -d to inspect any issues that might occur
hope this helps.

**Comment by vicXvac** [comment]
Another thing that helped for me was changing the docker context to use the default instead of docker desktop

**Reply by g1ronn1mo** [reply]
How?

**Reply by vicXvac** [reply]
I installed docker desktop via the official docker desktop documentations, once installed the docker that I was using was docker desktop, u can check which docker context ur using in the docker desktop settings then check which ur using, am running parrot OS with gnome DE so my contexts were
1, docker-desktop
2, default

So the one that was running was docker desktop
U can also check which docker context ur using by running the command

`docker context ls`

It shows the context of docker ur using and if u want to change the context u can run the command 

`docker context use <context-name>`

I started using the default context and the vector thing was working fine

Though after it was working fine the supabase db diff command or anything else to do with creating the shadow database was failing so I cloned the supabase cli repo and made the change to the health time out from 2 minutes to 4 minutes then installed this custom cli on my parrot os then everything was dope dope
